File: app/src/main/java/com/example/javademogithubpractice/mvp/contract/IBaseContract.java


package com.example.javademogithubpractice.mvp.contract;

import android.app.ProgressDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;


public interface IBaseContract {

    interface View {
        void showProgressDialog(String content);
        void dismissProgressDialog();
        ProgressDialog getProgressDialog(String content);
        void showTipDialog(String content);
        void showConfirmDialog(String msn, String title, String confirmText, DialogInterface.OnClickListener confirmListener);
        void showToast(String message);
        void showInfoToast(String message);
        void showSuccessToast(String message);
        void showErrorToast(String message);
        void showWarningToast(String message);
        default void showLoading() {}
        default void hideLoading() {}
        void showLoginPage();
    }

    interface Presenter<V extends IBaseContract.View>{
        void onSaveInstanceState(Bundle outState);
        void onRestoreInstanceState(Bundle outState);
        void attachView(@NonNull V view);
        void detachView();
        void onViewInitialized();
        @Nullable
        Context getContext();
    }
}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/mvp/contract/IBaseContract.kt
package com.example.kotlindemogithubproject.mvp.contract

import android.app.ProgressDialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.annotation.NonNull

interface IBaseContract {

    interface View {

        fun showProgressDialog(content: String)

        fun dismissProgressDialog()

        fun getProgressDialog(content: String): ProgressDialog

        fun showTipDialog(content: String)

        fun showConfirmDialog(
            msn: String,
            title: String,
            confirmText: String,
            confirmListener: DialogInterface.OnClickListener
        )

        fun showToast(message: String)

        fun showInfoToast(message: String)

        fun showSuccessToast(message: String)

        fun showErrorToast(message: String)

        fun showWarningToast(message: String)

        fun showLoading()

        fun hideLoading()

        fun showLoginPage()

    }

    interface Presenter<V> {

        val context: Context

        fun onSaveInstanceState(outState: Bundle)

        fun onRestoreInstanceState(outState: Bundle)

        fun attachView(@NonNull view: V)

        fun detachView()

        fun onViewInitialized()
    }

}
